plugins/modules/system/java_cert.py

```diff
@@ -185,21 +193,109 @@ def check_cert_present(module, executable, keystore_path, keystore_pass, alias,
     test_cmd = ("%s -noprompt -list -keystore '%s' -storepass '%s' "
                 "-alias '%s' %s") % (executable, keystore_path, keystore_pass, alias, get_keystore_type(keystore_type))
 
-    check_rc, dummy, dummy = module.run_command(test_cmd)
+    (check_rc, dummy, dummy) = module.run_command(test_cmd)
     if check_rc == 0:
         return True
     return False
 
 
-def import_cert_url(module, executable, url, port, keystore_path, keystore_pass, alias, keystore_type, trust_cacert):
-    ''' Import certificate from URL into keystore located at keystore_path '''
+def get_digest_from_x509_in_jks_store(module, executable, keystore_path,
+                                      keystore_pass, cert_alias, keystore_type, digest):
+    """ Exports an X509 cert from a JKS store, and returns the digest of that cert """
+    cert_pem = _export_cert(module, executable, keystore_path,
+                            keystore_pass, cert_alias, keystore_type)
+    cert_digest = _convert_pem_string_to_x509_object(cert_pem).digest(digest)
+    return cert_digest
+
+
+def _export_cert(module, executable, keystore_path, keystore_pass, alias, keystore_type):
+    """ Uses keytool to export the certificate with the specified alias in PEM.
+          The certificate is returned as a PEM formatted string """
+    export_cmd = ("%s -noprompt -rfc -export -keystore '%s' -storepass '%s' "
+                  "-alias '%s' %s") % (executable, keystore_path, keystore_pass, alias, get_keystore_type(keystore_type))
```

Comment:
Create a list of arguments instead of a string that has to be interpreted by the shell:
```suggestion
    export_cmd = [
        executable,
        "-noprompt",
        "-rfc",
        "-export",
        "-keystore",
        keystore_path,
        "-storepass"
        keystore_pass,
        "-alias",
        alias,
        get_keystore_type(keystore_type)
    ]
```
Also, passing in passwords on the command line is a security issue and should not be done. The standard way is to pass it in via stdin.

(Same for other cases where you use `run_command`.)